src/main/java/org/beanpod/switchboard/service/StreamLogService.java

```diff
@@ -27,8 +34,25 @@ public StreamLogDto createLog(
 
     streamLog.setLogEntity(logEntity);
 
-    StreamLogDto streamLogDto = logStreamMapper.toLogStreamDto(streamLog);
+    StreamLogDto streamLogDto = streamLogMapper.toLogStreamDto(streamLog);
 
     return streamLogDao.createStreamLog(streamLogDto);
   }
+
+  public StreamLogDto createLog(CreateStreamLogRequest createStreamLogRequest) {
+    Optional<StreamDto> streamDto =
+        Optional.of(Long.valueOf(createStreamLogRequest.getStreamId()))
+            .map(streamDao::getStreamById);
+
+    if (!streamDto.isPresent()) {
+      return null;
+    }
+
+    return createLog(
+        logMapper.map(createStreamLogRequest.getDateTime()),
+        createStreamLogRequest.getMessage(),
+        streamDto.get().getInputChannel().getDecoder().getSerialNumber(),
+        streamDto.get().getOutputChannel().getEncoder().getSerialNumber(),
+        createStreamLogRequest.getStreamId().toString());
+  }
```

Comment:
You may use functional style as suggested below to make your method slightly more compact and more uniform rather than breaking it up with a if-statement.
```
public StreamLogDto createLog(CreateStreamLogRequest createStreamLogRequest) {
    return Optional.of(Long.valueOf(createStreamLogRequest.getStreamId()))
        .map(streamDao::getStreamById)
        .map(
            dto ->
                createLog(
                    logMapper.map(createStreamLogRequest.getDateTime()),
                    createStreamLogRequest.getMessage(),
                    dto.getInputChannel().getDecoder().getSerialNumber(),
                    dto.getOutputChannel().getEncoder().getSerialNumber(),
                    createStreamLogRequest.getStreamId().toString()))
        .orElse(null);
  }
```

Reply:
Thanks for the suggestions. I'll make sure to apply them before merging next PR